python/ray/_private/worker.py

```diff
@@ -881,9 +881,11 @@ def deserialize_objects(self, serialized_objects, object_refs):
                 continue
 
             object_id = obj_ref.hex()
-            out_of_band_tensors[
-                object_id
-            ] = self.gpu_object_manager.get_out_of_band_tensors(object_id)
+            if object_id not in gpu_objects:
+                gpu_objects[object_id] = self.gpu_object_manager.get_gpu_object(
+                    object_id
+                )
+            gpu_objects[object_id].num_readers += 1
```

Comment:
We didn't know `num_readers` when we construct `GPUObject` because there are two places that we construct it:
* (1) serialize
* (2) `__ray_recv__`

For (1), we didn't know `num_readers` at that moment.